### PR TITLE
group init flag defines into an enum

### PR DESCRIPTION
**Pull request by Lokathor**
## Description
This groups the init flags into a named enum, similar to other flags groupings in the library.

I hope I didn't miss any other locations where the function signatures need to be updated! I'm not an expert on the library's organization though.

## Existing Issue(s)
Closes https://github.com/libsdl-org/SDL/issues/7133

### CODE REVIEW

**Comment by Lokathor** [comment]
Ah, looks like the dynapi needs to be updated too? But running the `gendynapi.py` script doesn't seem to touch any files, so I'm unsure what's going on there.

File: include/SDL3/SDL_init.h [hunk 1]
```diff
@@ -39,25 +39,24 @@ extern "C" {
 /* As of version 0.5, SDL is loaded dynamically into the application */
 
 /**
- *  \name SDL_INIT_*
- *
- *  These are the flags which may be passed to SDL_Init().  You should
- *  specify the subsystems which you will be using in your application.
+ * These are the flags which may be passed to SDL_Init().  You should
+ * specify the subsystems which you will be using in your application.
```
Comment (madebr) [inline include/SDL3/SDL_init.h:43]: I briefly looked at other enums and they have a *brief* doc string + a longer one.
I would also add `\sa`, but I don't know whether the sdl wiki supports that on typedef'ed enums.

```suggestion
 *   \brief Initialization flags for SDL_Init and/or SDL_InitSubSystem
 *
 * These are the flags which may be passed to SDL_Init().  You should
 * specify the subsystems which you will be using in your application.
 *
 * \sa SDL_Init
 * \sa SDL_InitSubSystem
```

File: include/SDL3/SDL_init.h [hunk 2]
```diff
@@ -105,7 +104,7 @@ extern "C" {
  * \sa SDL_SetMainReady
  * \sa SDL_WasInit
  */
-extern DECLSPEC int SDLCALL SDL_Init(Uint32 flags);
+extern DECLSPEC int SDLCALL SDL_Init(SDL_InitFlags flags);
```
Comment (slouken) [inline include/SDL3/SDL_init.h:107]: This should stay Uint32, as it's a bitmask of flags, and compilers tend to complain if you try to stuff OR'd enums into an enum.

File: include/SDL3/SDL_init.h [hunk 3]
```diff
@@ -39,8 +39,13 @@ extern "C" {
 /* As of version 0.5, SDL is loaded dynamically into the application */
 
 /**
+ *   \brief Initialization flags for SDL_Init and/or SDL_InitSubSystem
+ *
  * These are the flags which may be passed to SDL_Init().  You should
  * specify the subsystems which you will be using in your application.
+ *
+ * \sa SDL_Init
+ * \sa SDL_InitSubSystem
```
Comment (slouken) [inline include/SDL3/SDL_init.h:48]: You should include SDL_QuitSubSystem here as well.

**Comment by slouken** [comment]
Looks good, thanks!

**Pull request merged**